icu4c/source/test/fuzzer/fuzzer_driver.cpp

```diff
@@ -10,11 +12,48 @@ extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size);
 
 int main(int argc, char* argv[])
 {
-  (void) argc;
-  (void) argv;
-  const char *fuzzer_data = "abc123";
-   
-  LLVMFuzzerTestOneInput((const uint8_t *) fuzzer_data, strlen(fuzzer_data));
+  bool build_under_sanitizer = false;
+#if defined(__has_feature)
```

Comment:
FWIW, there's a wrapper macro for the compiler feature test [here in platform.h](https://github.com/unicode-org/icu/blob/20c29becd69b19c62cb6618665fd773eeb73dab1/icu4c/source/common/unicode/platform.h#L447), so it should already be included by other headers.

So this could possibly be something like:
```c++
#if UPRV_HAS_FEATURE(address_sanitizer)
   build_under_sanitizer = true;
#endif
```